File: Practica1_i-Lunch/src/gestorAplicacion/usuariosRestaurante/Cliente.java
package gestorAplicacion.usuariosRestaurante;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.Random;
import java.io.Serializable;

import gestorAplicacion.gestionRestaurante.*;

public class Cliente implements Serializable {
	
	// Serializacion
	private static final long serialVersionUID = 1L;
	private static ArrayList<Cliente> clientes;
	static {
		clientes = new ArrayList<Cliente>();
	}
	
	// Atributos
	private int telefono;
	private String primerNombre;
	private String segundoNombre;
	private String primerApellido;
	private String segundoApellido;
	private String direccion;
	private LocalDateTime fechaNacimiento;
	

	public Cliente(int telefono, String primerNombre, String segundoNombre, String primerApellido,
			String segundoApellido, String direccion, LocalDateTime fechaNacimiento, Pedido pedidoActivo,
			ArrayList<Pedido> historialPedidos) {
		super();
		this.telefono = telefono;
		this.primerNombre = primerNombre;
		this.segundoNombre = segundoNombre;
		this.primerApellido = primerApellido;
		this.segundoApellido = segundoApellido;
		this.direccion = direccion;
		this.fechaNacimiento = fechaNacimiento;
		this.pedidoActivo = pedidoActivo;
		this.historialPedidos = historialPedidos;
		clientes.add(this);
	}
	
	public Cliente() {
		this(0, "", "", "", "", "", null, null, null);
	}

	// Aca se almacenara el pedido que este la persona realizando en el momento,
	// al despachar o cancelar dicho pedido este atributo tiene que volver a null.
	// y en particular despues de ser despachado este pedido activo tiene que ser agregado al historial.
	private Pedido pedidoActivo = null;
	
	// ArrayList en el cual se agregaran todos los pedidos hechos por el cliente a lo largo del tiempo.
	private ArrayList<Pedido> historialPedidos = new ArrayList<Pedido>();
	
	////////////// METODOS ////////////////
	public Pedido iniciarPedido() {
		return new Pedido(this, new Random().nextInt(999999),"Ni idea, falta implementar", LocalDateTime.now(), null);		
	}
	
	public void agregarProducto(Producto producto) {
		pedidoActivo.getProductos().add(producto);
	}
	
	public void eliminarProducto(Producto producto) {
		pedidoActivo.getProductos().remove(producto);
	}
	
	public void pedir(Pedido pedido, String tipo, String mensaje) {
		//PENDIENTE
	}
	
	///////// GETTERS AND SETTERS /////////
	public int getTelefono() {
		return telefono;
	}
	public void setTelefono(int telefono) {
		this.telefono = telefono;
	}
	public String getPrimerNombre() {
		return primerNombre;
	}
	public void setPrimerNombre(String primerNombre) {
		this.primerNombre = primerNombre;
	}
	public String getSegundoNombre() {
		return segundoNombre;
	}
	public void setSegundoNombre(String segundoNombre) {
		this.segundoNombre = segundoNombre;
	}
	public String getPrimerApellido() {
		return primerApellido;
	}
	public void setPrimerApellido(String primerApellido) {
		this.primerApellido = primerApellido;
	}
	public String getSegundoApellido() {
		return segundoApellido;
	}
	public void setSegundoApellido(String segundoApellido) {
		this.segundoApellido = segundoApellido;
	}
	public String getDireccion() {
		return direccion;
	}
	public void setDireccion(String direccion) {
		this.direccion = direccion;
	}
	public LocalDateTime getfechaNacimiento() {
		return fechaNacimiento;
	}
	public void setfechaNacimiento(LocalDateTime fechaNacimiento) {
		this.fechaNacimiento = fechaNacimiento;
	}

	public static ArrayList<Cliente> getClientes() {
		return clientes;
	}

	public Pedido getPedidoActivo() {
		return pedidoActivo;
	}

	public void setPedidoActivo(Pedido pedidoActivo) {
		this.pedidoActivo = pedidoActivo;
	}

	public ArrayList<Pedido> getHistorialPedidos() {
		return historialPedidos;
	}

	public void setHistorialPedidos(ArrayList<Pedido> historialPedidos) {
		this.historialPedidos = historialPedidos;
	}	
}

File: Practica2_i-Lunch/src/gestorAplicacion/usuariosRestaurante/cliente.py
from gestorAplicacion.usuariosRestaurante.usuario import Usuario

class Cliente(Usuario):

    # Atributos estaticos

    _clientes = []

    # Constructor

    def __init__(self,telefono = None, nombre = None, direccion = None, edad = None, pedidoActivo = None, correoElectronico = None):
        super().__init__()
        self._telefono = telefono
        self._nombre = nombre
        self._direccion = direccion
        self._edad = edad
        self._correoElectronico = correoElectronico
        self._pedidoActivo = pedidoActivo
        self._historialPedidos = []

        
        Cliente._clientes.append(self)

    # Getters y Setters
    def getTelefono(self):
        return self._telefono
    def setTelefono(self,telefono):
        self._telefono = telefono
    def getNombre(self):
        return self._nombre
    def setNombre(self, nombre):
        self._nombre = nombre
    def getDireccion(self):
        return self._direccion
    def setDireccion(self,direccion):
        self._direccion = direccion
    def getEdad(self):
        return self._edad
    def setEdad(self, edad):
        self._edad  = edad
    def getCorreoElectronico(self):
        return self._correoElectronico
    def setCorreoElectronico(self, correo):
        self._correoElectronico = correo
    def getHistorialPedidos(self):
        return self._historialPedidos
    def setHistorialPedidos(self, historial):
        self._historialPedidos = historial

    @classmethod
    def getClientes(cls):
        return cls._clientes

    @classmethod
    def setClientes(cls, clientes):
        cls._clientes = clientes

    # Metodos



    # Implementacion de la interfaz

    def informacion(self):
        return f'El cliente {self._nombre} con email {self._correoElectronico} y telefono {self._telefono}' \
               f'vive en la direccion {self._direccion} y tiene {self._edad} años.\n Ha hecho {len(self._historialPedidos)}' \
               f' pedidos en la aplicacion'

    def agregarPedidoHistorial(self,  pedido):
        self._historialPedidos.append(pedido)